target/linux/ar71xx/config-4.9

```diff
@@ -12,6 +12,7 @@ CONFIG_ARCH_HAS_ELF_RANDOMIZE=y
 CONFIG_ARCH_HAS_RESET_CONTROLLER=y
 # CONFIG_ARCH_HAS_SG_CHAIN is not set
 CONFIG_ARCH_HIBERNATION_POSSIBLE=y
+CONFIG_ARCH_MMAP_RND_BITS=16
```

Comment:
This should go into generic, what about kernel 4.14 and what about other architectures?

Reply:
I can easily move patches to target/linux/generic but ARCH_MMAP_RND_BITS depends on architecture.

It must be between ARCH_MMAP_RND_BITS_MIN and  ARCH_MMAP_RND_BITS_MAX. Actually _MAX is:
- 16 on x86
- 32 on x86_64
- 14, 15  or 16 on arm depending on page size
- 14, 16, 19, 24, 27, 29, 30, 31, 33 on arm64 depending on pagesize and VA space.
- 16 on mips
- 18 on mips64

target/linux/generic/config-4.XX defines it to 8, a really low value, lower than min on some architectures. It might be a problem if it's not overridden in architecture configs.

Kernel defaults to ARCH_MMAP_RND_BITS_MIN. As OpenWrt is designed for small memory footprint, memory fragmentation should not be a problem. A way to improve this is to add a kernel patch that change default behavior to MAX.

Do you see a better way to do it ? I don't see an easy way to handle it in OpenWrt config system.
```
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -613,7 +613,7 @@
 	int "Number of bits to use for ASLR of mmap base address" if EXPERT
 	range ARCH_MMAP_RND_BITS_MIN ARCH_MMAP_RND_BITS_MAX
 	default ARCH_MMAP_RND_BITS_DEFAULT if ARCH_MMAP_RND_BITS_DEFAULT
-	default ARCH_MMAP_RND_BITS_MIN
+	default ARCH_MMAP_RND_BITS_MAX
 	depends on HAVE_ARCH_MMAP_RND_BITS
 	help
 	  This value can be used to select the number of bits to use to
```

Reply:
Here is says it should be 3 to 4 bites less than the maximum: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d07e22597d1d355829b7b18ac19afa912cf758d1 Normally the address space is not fully used on the typical devices using OpenWrt., You can set ARCH_MMAP_RND_BITS per target to a better value, the default 8 is alos only taken when the architecture allows this.

I think we should set this on a per target base to a better default.